### ISSUE TITLE
Instance for `TVar`

### ISSUE THREAD

**Issue by hvr** [haskell/deepseq#15]
This is a left-over issue from #6

As described in https://github.com/haskell/deepseq/issues/6#issuecomment-147619390 by @glguy defining an instance for `TVar` is complicated since we'd lose `{-# LANGUAGE Safe #-}` by import `TVar` which seems undesirable, given all other instances can be imported safely.

**Comment by phadej** [comment]
should the instance live in `stm` package instead? No safety problem then?

**Comment by hvr** [comment]
@phadej yeah, the only not-so-nice  is that in `stm` it'd be an orphan

**Comment by treeowl** [comment]
One option would be to add a `Trustworthy` module to define the `rnf` function and import it into `Control.DeepSeq`. But that seems pretty silly! The right answer is surely to add a safe `TVar` re-export somewhere in `base`.

**Comment by RyanGlScott** [comment]
> The right answer is surely to add a safe `TVar` re-export somewhere in `base`.

That would be nice. But this needn't stop us from adding an `NFData` instance for `TVar` today, since there's a module [`Control.DeepSeq.BackDoor`](https://github.com/haskell/deepseq/blob/18fc08e66d2ad1a72a6f78708acbead007d4a580/Control/DeepSeq/BackDoor.hs) specifically for the purpose of safely re-exporting things from `base` that currently can't be imported safely. We'd likely have to perform some surgery on the CPP bounds to accommodate `TVar`, but it should be possible.

**Comment by hvr** [comment]
Yeah, this ticket predates the backdoor module by 1 year... so I mostly forgot about this... if anyone wants to take a stab at a PR...? :-)

**Comment by spacekitteh** [comment]
Anyone?

**Comment by mixphix** [comment]
Pull requests are welcome.
